design/open/lakefs-on-kv-testing-plan.md

```diff
@@ -0,0 +1,128 @@
+# lakeFS on KV - Testing Plan
+
+## Global Testing
+
+### Migration
+* Global tests to verify and benchmark the migration process
+* Need to verify that the system behavior is not affected by the move to the KV DB
+* [Question] - Do we need to dispose of these once migration to KV is globally completed?
+
+#### Operational Testing
+* Set of tests that will run DB migration as part of the test
```

Comment:
How is this different from the previous section?

Reply:
The previous section is about looking at the DB before and after the migration, and comparing its context.
This section is about having certain operations running (e.g. several multipart uploads), then, as part of the test, a migration to KV takes place (probably need to stop lakeFS, migrate, verify it was successful (?) and start lakeFS), then continue the operations where we left (in that case, resume some and some complete multipart uploads, as well as starting new ones) and verify it works as expected.
I agree that it seems quite the same, as the mutlipart uploads DB usage is very basic, but I believe that once we get into modules that use the DB in a more complexed way (e.g. `pkg/auth`) it will be beneficial.
This step is about building the infrastructure for such test and not necessarily about the `multipart uploads` aspect

Reply:
I think the current direction is to do the "migration" by reading from a dump so I don't think we will require to start and stop lakeFS for this flow - just prepare some DB dump with data ahead of running the test.

Reply:
@N-o-Z How do we make sure we are not missing any DB updates arriving after the dump and before KV done importing?

Reply:
@itai-david I think the data should really be scenario dependant. For example if you want to test migration during multipart upload - you can build the dump file in a way that continuing the multipart upload after migration you will expect a certain outcome (upload successful + file content)